FlickrAssignment/FlickrViewModel.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  FlickrViewModel.swift
+//  FlickrAssignment
+//
+//  Created by admin on 15/03/2022.
+//
+
+import Foundation
+
+protocol FlickrModelType: AnyObject {
+    func updateImageResponce(responce: Pictures)
+}
+
+class FlickrViewModel: FlickrModelType {
+    
+    weak var delegate: Result?
+    init(delegate: Result){
+        self.delegate = delegate
+    }
+    
+    
+    let networkManager = flickrNetworkClass()
```

Comment:
use protocol type instead of class type,  use dependency inversion principle.
use better name for flickrNetworkClass and  class name should be capital